framework/base/Model.php

```diff
@@ -216,7 +216,7 @@ public function scenarios()
         }
 
         foreach ($scenarios as $scenario => $attributes) {
-            if (empty($attributes) && $scenario !== self::SCENARIO_DEFAULT) {
+            if ($scenario !== self::SCENARIO_DEFAULT && empty($attributes)) {
```

Comment:
Well, yes. I've started from the ones acting with files which is significant and then it was hard to stop :)